### PR TITLE
fix: do not block on netcheck

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Review by rklaehn — APPROVED** [review]
I don't know the details, but seems like an improvement...

File: iroh-net/src/magicsock.rs [hunk 1]
```diff
@@ -1097,10 +1105,37 @@ impl EndpointUpdateState {
         }
     }
 
+    /// Schedules a new run, either starting it immediately if none is running or
+    /// scheduling it for later.
+    fn schedule_run(&mut self, why: &'static str) {
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply iroh-net/src/magicsock.rs:1110]: could be an enum, but it is only used for logging, which is why I never bothered

[the remark this answers is not in the document]
Comment (rklaehn) [reply iroh-net/src/magicsock.rs:1110]: Yes, it is not that important. But I find an enum much nicer when trying to make sense of the code.

[the remark this answers is not in the document]
Comment (Frando) [reply iroh-net/src/magicsock.rs:1110]: Yes I also think an enum makes sense. But can be a followup too I guess.

**Review by Frando — APPROVED** [review]
Yes, this is much much better :)

**Pull request merged**